### PR TITLE
configs: Add configurable GPU L1,L2 num banks and L2 latencies

### PR DESCRIPTION
**Pull request by v-ramadas**
Previously, the L1, L2 number of banks and L2 latencies were not configurable through command line arguments. This commit adds support to configure them through the arguments '--tcp-num-banks' for number of banks in L1, '--tcc-num-banks' for number of banks in L2, and '--tcc-tag-access-latency', and '--tcc-data-access-latency'

Change-Id: Ie3b713ead16865fd7120e2d809ebfa56b69bc4a1

### CODE REVIEW

**Review by mattsinc — APPROVED** [review]
This looks good to me, but I'm not sure if @powerjg has checked in his ViperBoard.  If so, we'll need to work with him to integrate these into that.

**Review by abmerop — APPROVED** [review]

File: configs/ruby/GPU_VIPER.py
Comment (powerjg) [inline configs/ruby/GPU_VIPER.py:308]: Isn't this logic backwards?

**Review by powerjg — COMMENTED** [review]
I'm generally against adding command line options. I think they are easy to misuse and easy to make mistakes.

That said, I won't block this change.

**Comment by powerjg** [comment]
> This looks good to me, but I'm not sure if @powerjg has checked in his ViperBoard. If so, we'll need to work with him to integrate these into that.

We haven't gotten that in yet.

**Comment by mattsinc** [comment]
> > This looks good to me, but I'm not sure if @powerjg has checked in his ViperBoard. If so, we'll need to work with him to integrate these into that.
> 
> We haven't gotten that in yet.

Until that is merged in, I'm not sure what the alternative would be?  I guess just enabling the underlying code added here but not having a command line arg?

File: configs/ruby/GPU_VIPER.py
Comment (mattsinc) [reply configs/ruby/GPU_VIPER.py:308]: Sorry, I missed that there was an important, pending question here.  I'm not sure what the issue is though, both options are assigned to the corresponding one?

**Pull request merged**
